packages/react/src/chat.react.ts

```diff
@@ -71,7 +62,7 @@ class ReactChatState<UI_MESSAGE extends UIMessage>
   replaceMessage = (index: number, message: UI_MESSAGE) => {
     this.#messages = [
       ...this.#messages.slice(0, index),
-      message,
+      this.snapshot(message),
```

Comment:
since there is no test, can you add a comment here?

Reply:
yes can do!